### ISSUE TITLE
Make sure overlay prompts are dismissible using keyboard navigation

### ISSUE THREAD

**Issue by aschweigert** [Automattic/newspack-popups#849]
Reported by Calmatters: https://newspack-pub.slack.com/archives/G0122C8BVPC/p1649981164117259

Any prompt that is displayed over content (popups, toasters, etc) should be dismissible using the Esc key in addition to clicking on the X or outside of the body of the prompt.
